Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py

```diff
@@ -198,6 +201,42 @@ def _fork_remote(cls, origin, username, name):
             return '{}:{}/{}.git'.format(origin.split(':')[0], username, name)
         return None
 
+    @classmethod
+    def _security_levels(cls, repository, username=None):
+        proc = run(
+            [local.Git.executable(), 'config', '--get-regexp', 'webkitscmpy.remotes'],
+            capture_output=True, cwd=repository.root_path,
+            encoding='utf-8',
+        )
+        if proc.returncode:
+            return {}
+
+        level = 0
+        result = {}
+        for line in proc.stdout.splitlines():
+            key, value = line.split(' ', 1)
+            match = cls.REMOTE_RE.match(value)
+            if match:
+                result['{}:{}'.format(match.group('host'), match.group('path'))] = level
+                if username:
+                    team, repository = match.group('path').split('/', 1)
+                    if team == repository:
+                        forked_name = repository
+                    elif '-' in repository:
+                        forked_name = repository
+                    else:
+                        forked_name = '{}-{}'.format(repository, team)
+                    result['{}:{}/{}'.format(match.group('host'), username, forked_name)] = level
+
+            result[str(key.split('.')[-1])] = level
+            if username:
+                if key.split('.')[-1] == 'origin':
+                    result['fork'] = level
+                else:
+                    result['{}-fork'.format(key.split('.')[-1])] = level
+            level += 1
```

Comment:
Is the security level really just based on the order in which the remotes are listed in a `.git/config` file, or is there some magic that happens with `git config --get-regexp webkitscmpy.remotes` to ensure the correct order each time it is run?

Maybe I'm not understanding the threat model, though.

Reply:
These aren't listed in `.git/config`, they come from `metadata/git_config_extension`. It is the order in this file (which is checked in) that matters. The nice thing about this approach is that it can allow other workflows to define additional remotes in `.git/config` and still have this logic work, but we can discuss alternatives here (like perhaps defining security levels in `metadata/git_config_extension` if this implicit order concept isn't making sense to folks.

Reply:
I know there is precedent for this pattern in the code, but I am scared at how magical the ordering is. It seems like any accidental reordering could lead to a leak of security content.

Would you mind putting a comment in `metadata/git_config_extension` explaining the significance of the ordering?

Reply:
@emw-apple and I discussed this offline, instead of relying on implicit ordering here, we're going to do https://github.com/WebKit/WebKit/pull/11251, which makes the security level an explicit part of the configuration. Once that PR lands, I'll adopt it here.